coreLib/src/main/java/dev/kosmx/playerAnim/api/layered/AnimationStack.java

```diff
@@ -88,4 +91,24 @@ public boolean removeLayer(int layerLevel) {
         return layers.removeIf(integerIAnimationPair -> integerIAnimationPair.getLeft() == layerLevel);
     }
 
+    @Nullable
+    public FirstPersonAnimation getActiveFirstPersonAnimation(float tickDelta) {
+        for (Pair<Integer, IAnimation> layer : layers) {
+            IAnimation animation = layer.getRight();
+            if (animation instanceof ModifierLayer) {
+                animation = ((ModifierLayer)animation).getAnimation();
+            }
```

Comment:
This works for my cases, but doesn't look like a good universal solution.

Reply:
I'll definitely refactor this.